runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientCreateTopicsFactory.java

```diff
@@ -0,0 +1,1676 @@
+                break;
+            default:
+                delegate.cleanupApplication(traceId, errorCode);
+                doNetworkEnd(traceId, authorization);
+                break;
+            }
+        }
+
+        @Override
+        protected void onDecodeSaslAuthenticateResponse(
+            long traceId,
+            long authorization,
+            int errorCode)
+        {
+            switch (errorCode)
+            {
+            case ERROR_NONE:
+                encoder = encodeCreateTopicsRequest;
+                decoder = decodeCreateTopicsResponse;
+                break;
+            default:
+                delegate.cleanupApplication(traceId, errorCode);
+                doNetworkEnd(traceId, authorization);
+                break;
+            }
+        }
+
+        @Override
+        protected void onDecodeSaslResponse(
+            long traceId)
+        {
+            nextResponseId++;
+            signaler.signalNow(originId, routedId, initialId, traceId, SIGNAL_NEXT_REQUEST, 0);
+        }
+
+        private void onDecodeCreateTopicsResponse(
+            long traceId,
+            long authorization,
+            int throttle,
+            List<CreateTopicsResponse> topics)
+        {
+            delegate.doApplicationBegin(traceId, authorization, throttle, topics);
+        }
+
+        private void cleanupNetwork(
+            long traceId)
+        {
+            doNetworkResetIfNecessary(traceId);
+            doNetworkAbortIfNecessary(traceId);
+
+            delegate.cleanupApplication(traceId, ERROR_NONE);
+        }
+
+        private void cleanupDecodeSlotIfNecessary()
+        {
+            if (decodeSlot != NO_SLOT)
+            {
+                decodePool.release(decodeSlot);
+                decodeSlot = NO_SLOT;
+                decodeSlotOffset = 0;
+                decodeSlotReserved = 0;
+            }
+        }
+
+        private void cleanupEncodeSlotIfNecessary()
+        {
+            if (encodeSlot != NO_SLOT)
+            {
+                encodePool.release(encodeSlot);
+                encodeSlot = NO_SLOT;
+                encodeSlotOffset = 0;
+                encodeSlotTraceId = 0;
+            }
+        }
+
+        private void cleanupBudgetIfNecessary()
+        {
+            if (initialDebIndex != NO_DEBITOR_INDEX)
+            {
+                initialDeb.release(initialDebIndex, initialId);
+                initialDebIndex = NO_DEBITOR_INDEX;
+            }
+        }
+    }
+
+    private final class CreateTopicsRequest
+    {
+        private final List<Topic> topics;
+        private final int timeoutMs;
+        private final byte validateOnly;
+
+        private CreateTopicsRequest(
+            List<Topic> topics,
+            int timeoutMs,
+            byte validateOnly)
+        {
+            this.topics = topics;
+            this.timeoutMs = timeoutMs;
+            this.validateOnly = validateOnly;
+        }
+    }
+
+    private final class Topic
```

Comment:
Suggest we call these `Info` classes, like `TopicInfo`, `AssignmentInfo`, etc.
Also these seem to be more like `record` types than `class` types.
Do they need to be instance level inside the factory, or can they be static?
Note: inner `record` is implicitly `static`.

Reply:
> Do they need to be instance level inside the factory, or can they be static?
not really but I thought since they are not used in other classes it should be self-contained.